coderd/database/databasefake/databasefake.go

```diff
@@ -1743,6 +1743,9 @@ func (q *fakeQuerier) GetPreviousTemplateVersion(_ context.Context, arg database
 		if templateVersion.ID == currentTemplateVersion.ID {
 			continue
 		}
+		if templateVersion.TemplateID != currentTemplateVersion.TemplateID {
+			continue
+		}
```

Comment:
Ahh sorry for that. It was but right after I had to remove it because of some weird errors, my bad 🙏 

About the org check, it is being done to get the actual template version https://github.com/coder/coder/blob/5c141ba1a6d06cc06c05976878dff98db0081bd0/coderd/database/databasefake/databasefake.go#L1734-L1736 so when we check the `templateID`it is implicitly checking the org ID too. But I see value on making it explicitly. Going do that, thanks.

Reply:
Hmm, yes I see you are right about the org ID since `templates.id` is globally unique, I just prefer to be extra paranoid considering future changes. 😅. I'll leave it up to you to add it or not.